src/main/scala/io/iohk/ethereum/blockchain/sync/regular/BlockImporter.scala

```diff
@@ -249,8 +250,17 @@ class BlockImporter(
     }
   }
 
-  private def broadcastBlocks(blocks: List[Block], totalDifficulties: List[BigInt]): Unit = {
-    val newBlocks = (blocks, totalDifficulties).mapN(NewBlock.apply)
+  private def broadcastBlocks(
+      blocks: List[Block],
+      totalDifficulties: List[BigInt]
+  ): Unit = {
+    val constructNewBlock =
+      if (blocks.lastOption.exists(_.number < blockchainConfig.ecip1097BlockNumber))
+        NewBlock63.apply _
+      else
+        NewBlock64.apply(_, _, blockchain.getLatestCheckpointBlockNumber())
```

Comment:
Shouldn't the lastCheckpoint value be dependent on the block being broadcasted? For example, if we are broadcasting C1->B2->C3 then B2 should have C1 as the last checkpoint block.

At least for now the td of every new block broadcasted was giving the weight of that broadcasted block, not of the peer, if so the lastCheckpoint field should behave in the same way

Reply:
I thought that we're only broadcasting the top blocks. If that's not the case then yes, but this will be straight-forward after ETCM-263 using `blockchain.getChainWeightByHash`. Should I add a comment about fixing this in ETCM-263?

Reply:
Hmm if we were to follow how it's handled for the TD, wouldn't the lastCheckpointBlockNumber be part of the `BlockData` that's the result of the block execution? (instead of using `blockchain.getChainWeightByHash` as you suggest)

Independently of what we decide, shouldn't we handle the TD and lastCheckpointBlockNumber in the same way?

>Should I add a comment about fixing this in ETCM-263?

But yeah, it seems to be quite related to that handling, so we could include this as well for that task

Reply:
>  wouldn't the lastCheckpointBlockNumber be part of the BlockData that's the result of the block execution?

Actually `BlockData` will contain `ChainWeight`. 

We will handle this the same way once `ChainWeight` is fully introduced, so I need this to continue working on ETCM-263, and we'll need to ETCM-263 to properly implement this improvement